pset-functions.js

```diff
@@ -208,6 +318,31 @@ console.log('Problem 5 - bonus:')
 
 // Add your code below this line
 
+let words = []
+let counts = []
+let input = "Humpty Dumpty sat on a wall Humpty Dumpty had a great fall"
+let parsed = (input.replace(/[^a-zA-Z ]/g, "")).split(" ")
+
+if (parsed.length > 0) {
+   for (i=0; i<parsed.length; i++) {
+      parsed[i] = parsed[i].toLowerCase()
+   }
+}
+
+for (i=0; i<parsed.length; i++) {
+   if (!words.includes(parsed[i])) {
+      words.push(parsed[i])
+      counts.push(parseInt("1"))
+   } else {
+      let position = words.indexOf(parsed[i])
+      counts[position] = counts[position] + 1
+   }
+}
+
+for (i=0; i<words.length; i++) {
+console.log(words[i] + ": " + counts[i])
+}
```

Comment:
Lance, again great job taking edge cases into consideration by turning your parsed values in the array to lowercase letters throughout. Some minor suggestions, for line 324, no need to use the replace method, writing input.split(" ") works just fine here (so we can eliminate that extra step). Also, for line 343, you can use a template literal (ES6 upgrade) versus concatenating your string like so `${words[i]}: ${counts[i]}`. I noticed you used array's to push your values into; [here](https://github.com/jsd20200127/hw-02-functions-solution/blob/e87c29073cf7285a5045ec8708561f279474f2d2/pset-functions.js#L298) is alternate solution that uses an empty object instead (just for your reference; your solution works perfectly well). Really awesome job overall on this homework; knocked it out of the park! Keep up the great work :)